bitcoin/examples/sign-tx-taproot.rs

```diff
@@ -67,7 +67,7 @@ fn main() {
 
     // Sign the sighash using the secp256k1 library (exported by rust-bitcoin).
     let tweaked: TweakedKeypair = keypair.tap_tweak(None);
-    let signature = secp256k1::schnorr::sign(&sighash.to_byte_array(), tweaked.as_keypair());
+    let signature = secp256k1::schnorr::sign(&sighash.to_byte_array(), &tweaked.as_keypair().into_inner());
```

Comment:
OT but likely interesting to readers anyways. What is the reason that secp can schnorr sign any byte slice but ecdsa can only sign 32 bytes (I'm referring to the API here taking a slice and the ecdsa API taking an `Into<Message>`, found because I tried removing the `to_byte_array()` call)?